engine/core/src/main/java/org/datacleaner/metamodel/datahub/utils/JsonSchemasResponseParser.java

```diff
@@ -0,0 +1,262 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.metamodel.datahub.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.metamodel.schema.MutableTable;
+import org.datacleaner.metamodel.datahub.DataHubSchema;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+
+public class JsonSchemasResponseParser {
+
+    private static final Set<String> datastoreTypes = new HashSet<String>() {
+        {
+            add("GoldenRecordDatastore");
+            add("SourceRecordSourceFormatDatastore");
+            add("SourceRecordGoldenFormatDatastore");
+        }
```

Comment:
In effect this creates a subtype of HashSet in order to add values to it. It doesn't matter much because it's a single occurence but I do think it would be much more memory efficient and understandable to do:

```
new HashSet<>(Arrays.asList(("GoldenRecordDatastore", "SourceRecordSourceFormatDatastore","SourceRecordGoldenFormatDatastore"));
```